src/clusterfuzz/_internal/bot/fuzzers/centipede/engine.py

```diff
@@ -72,9 +78,68 @@ def _set_sanitizer_options(fuzzer_path):
   environment.set_memory_tool_options(sanitizer_options_var, sanitizer_options)
 
 
+def _parse_centipede_stats(
+    stats_file: str) -> Optional[Dict[str, Union[int, float]]]:
+  """Parses the Centipede stats file and returns a dictionary with labels
+  and their respective values.
+
+  Args:
+      stats_file: the path to Centipede stats file.
+
+  Returns:
+      a dictionary containing the stats.
+  """
+  if not os.path.exists(stats_file):
+    return None
+  with open(stats_file, 'r') as statsfile:
+    csvreader = csv.reader(statsfile)
+    l = list(csvreader)
+    # If the binary could not run at all, the file will be empty or with only
```

Comment:
should we also handle exceptions generally?

I'd expect there to be (hopefully rare) cases the file is malformed because centipede was killed in the middle of writing the file.

Reply:
Yes, I think you're right, let's be careful here and gracefully fail.